packages/playwright-test/src/runner.ts

```diff
@@ -307,7 +317,7 @@ export class Runner {
       const fileSuite = await this._loader.loadTestFile(file, 'runner');
       if (fileSuite._loadError)
         fatalErrors.push(fileSuite._loadError);
-      preprocessRoot._addSuite(fileSuite);
+      preprocessRoot._addSuite(fileSuite._deepClone());
```

Comment:
In theory, we can avoid cloning a) when calling the second time, b) when calling the first time but no `setupFiles` are defined.